src/Api/ItemDataProviderInterface.php

```diff
@@ -22,15 +21,12 @@
 interface ItemDataProviderInterface
 {
     /**
-     * Retrieves an item.
-     *
      * @param string      $resourceClass
      * @param string|null $operationName
      * @param int|string  $id
      * @param bool        $fetchData
      *
-     * @throws ResourceClassNotSupportedException
```

Comment:
I would keep this one explicitly because this is a case that must be handled in many cases (for others, it's fine to let them be converted to an error displayed to the end user).